qcodes/tests/dataset/test_experiment_container.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+import tempfile
+import os
+
+import qcodes as qc
+from qcodes.dataset.data_set import DataSet
+from qcodes.dataset.experiment_container import load_experiment_by_name, \
+    new_experiment
+from qcodes.dataset.sqlite_base import connect, init_db
+from qcodes.dataset.measurements import Measurement
+
+
+@pytest.fixture(scope="function")
+def empty_temp_db():
+    # create a temp database for testing
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        qc.config["core"]["db_location"] = os.path.join(tmpdirname, 'temp.db')
+        qc.config["core"]["db_debug"] = True
+        # this is somewhat annoying but these module scope variables
+        # are initialized at import time so they need to be overwritten
+        qc.dataset.experiment_container.DB = qc.config["core"]["db_location"]
+        qc.dataset.data_set.DB = qc.config["core"]["db_location"]
+        qc.dataset.experiment_container.debug_db = qc.config["core"]["db_debug"]
+        _c = connect(qc.config["core"]["db_location"], qc.config["core"]["db_debug"])
+        init_db(_c)
+        _c.close()
+        yield
+
+
+def test_run_loaded_experiment(empty_temp_db):
+    """
+    Test whether issue 996 as reported here:
```

Comment:
Again, I don't think we should duplicate git's work into the comments of the code.